pkg/cmd/util/clientcmd/factory.go

```diff
@@ -53,83 +55,115 @@ func DefaultClientConfig(flags *pflag.FlagSet) clientcmd.ClientConfig {
 // Factory provides common options for OpenShift commands
 type Factory struct {
 	*kubecmd.Factory
-	OpenShiftClientConfig clientcmd.ClientConfig
+	OpenShiftClientConfig ClientConfig
+}
+
+// intercept errors that need special treatment for end-users (e.g. no config file present)
+type ClientConfig struct {
+	KClientConfig clientcmd.ClientConfig
+}
+
+func (cfg *ClientConfig) ClientConfig() (*kclient.Config, error) {
+	c, err := cfg.KClientConfig.ClientConfig()
+	err = WrapClientErrors(err)
```

Comment:
Why wrap the errors, why not have `IsBlahError` directly check the error from the existing ClientConfig?